packages/jest-config/src/resolveConfigPath.ts

```diff
@@ -111,10 +118,23 @@ const findPackageJson = (pathToResolve: string) => {
   return undefined;
 };
 
-const hasPackageJsonJestKey = (packagePath: string) => {
-  const content = fs.readFileSync(packagePath, 'utf8');
+const resolveJestKey = (packageContent: any, packagePath: string) => {
+  if (hasPackageJsonJestKey(packageContent)) {
+    const {jest} = JSON.parse(packageContent);
```

Comment:
`JSON.parse()` is also called in `hasPackageJsonJestKey()`. Just a line above. Could this be somehow optimised?

Reply:
yes, i did some refactor to optimize it. Code already pushed.